### DISCUSSION TITLE
halo2 book 的 IPA 和 BCMS20 Appendix A.2的区别

### DISCUSSION THREAD

**Discussion by antyolkkkk** [z2o-k7e/halo2-step-by-step#33] [Q&A]
我在看halo2 book内积证明的这一节https://zcash.github.io/halo2/design/proving-system/comparison.html#bcms20-appendix-a2

这里不是很理解里面说的二者的区别以及原因，请问有没有同学有自己的理解可以分享~

**Comment by kurtpan666** [comment]
IPA发展脉络大致如此，具体比较我之后再补充: 
Zcash-Halo2-IPA 是 [BCMS20]-IPA 的变体；
[BCMS20]-IPA 是[BGH19]-Halo-IPA以及[BMM+19]-RIPP-IPA的generalization；
[BGH19]-Halo-IPA又是[BBB+18]-Bulletproofs-IPA的变体；
[BBB+18]-Bulletproofs-IPA又是[BCC+16]-IPA的变体。
